### PR TITLE
Fixed Issue #201

### PR DESCRIPTION
**Pull request by codingblazer**
Hey ,I have fixed the issue by implementing a alert dialog which restricts user if user gives same name as already existing file.Please review it. #201

### CODE REVIEW

**Comment by octacode** [comment]
@codingblazer
Hey, The code looks great. Don't you think It would be better to use AlertDialog box class instead of MaterialDialog box. It will actually make your PR independent of some external library:)
Thanks

**Comment by codingblazer** [comment]
@octacode Thanks for suggestion. Actually, MaterialDialog  has already been imported in the TemplateEditor.java and used at various places (also used in other files in project). Making another import for using AlertDialog in this file didn't looked good to me. Also using two different types of dialog in same file is not considered a good practice.Thanks

File: source-code/app/src/main/java/org/buildmlearn/toolkit/activity/TemplateEditor.java [hunk 1]
```diff
@@ -776,16 +784,34 @@ private String saveDraft() {
...
      */
     private void startSimulator() {
-        String filePath = saveProject();
-        if (filePath == null || filePath.equals("")) {
+        String message = saveProject();
+        if (message == null || message.equals("")) {
             Toast.makeText(this, "Build unsuccessful", Toast.LENGTH_SHORT).show();
             return;
         }
+        else if(message.equals("File already exists"))
```
Comment (codingblazer) [inline source-code/app/src/main/java/org/buildmlearn/toolkit/activity/TemplateEditor.java]: Null pointer Exception would never come as there is "if" statement above checking it. I am changing it however.

File: source-code/app/src/main/java/org/buildmlearn/toolkit/utilities/FileUtils.java [hunk 2]
```diff
@@ -1,7 +1,9 @@
 package org.buildmlearn.toolkit.utilities;
 
 import android.content.Context;
+import android.content.DialogInterface;
```
Comment (codingblazer) [inline source-code/app/src/main/java/org/buildmlearn/toolkit/utilities/FileUtils.java]: Removing the unused imports.

**Comment by opticod** [comment]
@codingblazer , can you please resolve the merge conflicts. 
Instead of showing them in alert box, it would be great if we show them the error beside the textview only. Validation will be performed before the project is saved. 
image
You can achieve this by ```textView.setError("message");```

**Comment by codingblazer** [comment]
@opticod Thanks for review. I have updated the branch as per review.

**Comment by opticod** [comment]
@codingblazer, Thanks for your changes, I am reviewing now. I was travelling this week, so was away from this.

**Comment by codingblazer** [comment]
@opticod Its working fine on my device. 
whatsapp image 2016-12-04 at 20 12 39
 Can you please recheck it or send me a screenshot ?

**Pull request merged**